File: src/main/java/com/tomtom/parkathon/rest/ParkingSpotsRestController.java
package com.tomtom.parkathon.rest;

import com.tomtom.parkathon.domain.ParkingSpot;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Arrays;
import java.util.List;

@RestController
public class ParkingSpotsRestController {

    @GetMapping("/parking-spots")
    public List<ParkingSpot> getAvailableParkingSpots() {
        return Arrays.asList(
                new ParkingSpot(52000000L, 13000000L),
                new ParkingSpot(52000000L, 13000000L),
                new ParkingSpot(52000000L, 13000000L));
    }
}

File: src/main/scala/com/tomtom/parkathon/rest/ParkingSpotsRestController.scala
package com.tomtom.parkathon.rest

import java.time.Instant
import java.util

import com.tomtom.parkathon.domain.ParkingSpot
import org.springframework.web.bind.annotation.{GetMapping, RestController}

import scala.collection.JavaConverters._

@RestController
class ParkingSpotsRestController {

  @GetMapping(Array("/parking-spots"))
  def getAvailableParkingSpots: util.List[ParkingSpot] =
    List(
      new ParkingSpot(52.133, 13.231, Instant.now),
      new ParkingSpot(52.132, 13.232, Instant.now),
      new ParkingSpot(52.131, 13.233, Instant.now),
    ).asJava
}
